crates/core/src/db/relational_db.rs

```diff
@@ -1449,4 +1473,61 @@ mod tests {
 
     //     Ok(())
     // }
+
+    #[test]
+    fn test_replay_corrupted_log() -> ResultTest<()> {
```

Comment:
It would be nice to also test the "safeguard" condition (log corrupted in the middle), but the default segment size in 1GiB. Punting on making this configurable at this point.